xaitk_saliency/utils/gen_coco_sal.py

```diff
@@ -256,8 +262,17 @@ def _coco_dets_to_formatted_mat(
         # assume class ids are arranged between 0 and the maximum id
         max_cat_id = max(coco_dset.cats.keys())
         class_mat = np.zeros((len(dets), max_cat_id+1))
+        objectness = np.ones(len(dets))
 
         for i, det in enumerate(dets):
-            class_mat[i, det['category_id']] = det['score']
 
-        return format_detection(bbox_mat, class_mat)
+            # look for class probabilities
+            if 'prob' in det:
```

Comment:
This seems a bit fragile- what if the user doesn't use the `prob` key to encode class probabilities? For example, it's conceivable that a user could still use the `score` key to store class probabilities, since the [official COCO documentation](https://cocodataset.org/#format-results) only mentions a `score` key. In this case, would it be safer to check the shape/length of the value instead of a specific key?

Reply:
`prob` is used by the [kwcoco schema](https://kwcoco.readthedocs.io/en/latest/_modules/kwcoco/coco_schema.html?highlight=prob), which is why I used it here. To me, it seems confusing to use `score` for the class probabilities given that the official COCO schema specifies it as float. Perhaps we could be more explicit in documentation about which key to use.

Reply:
Ok, that makes more sense now- please disregard my previous comment then. Thanks for the clarification!